packages/gatsby-theme-docs/src/templates/release-notes-list.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { graphql } from 'gatsby';
+import { MDXRenderer } from 'gatsby-plugin-mdx';
+import { MDXProvider } from '@mdx-js/react';
+import SpacingsStack from '@commercetools-uikit/spacings-stack';
+import { Markdown } from '@commercetools-docs/ui-kit';
+import LayoutReleaseNote from '../layouts/internals/layout-release-note';
+import LayoutReleaseNotesList from '../layouts/release-notes-list';
+import { SEO, ThemeProvider } from '../components';
+import markdownComponents from '../markdown-components';
+
+const releaseNoteMarkdownComponents = {
+  ...markdownComponents,
+  // NOTE: release notes content can only have headings starting from h4.
+  h1: Markdown.withAnchorLink(Markdown.H4),
+  h2: Markdown.withAnchorLink(Markdown.H5),
+  h3: Markdown.withAnchorLink(Markdown.H6),
+  h4: Markdown.withAnchorLink(Markdown.H6),
+};
+
+const ReleaseNotesListTemplate = (props) => (
+  <ThemeProvider>
+    <LayoutReleaseNotesList pageContext={props.pageContext}>
+      <Markdown.TypographyPage>
+        <SEO
+          title={props.pageContext.title}
+          excludeFromSearchIndex={props.pageContext.excludeFromSearchIndex}
+        />
+        <MDXProvider components={markdownComponents}>
+          <div>
+            <MDXRenderer>{props.data.mdx.body}</MDXRenderer>
+          </div>
+        </MDXProvider>
+        <div>
+          <MDXProvider components={releaseNoteMarkdownComponents}>
```

Comment:
And for the list of release notes content we use the modified list of markdown elements.